test/v1alpha1/service.go

```diff
@@ -345,3 +383,16 @@ func CheckServiceState(client *test.ServingAlphaClients, name string, inState fu
 func IsServiceReady(s *v1alpha1.Service) (bool, error) {
 	return s.Generation == s.Status.ObservedGeneration && s.Status.IsReady(), nil
 }
+
+// IsServiceNotReady will check the status conditions of the service and return false if the service is
+// ready.
+func IsServiceNotReady(s *v1alpha1.Service) (bool, error) {
+	return !s.Status.IsReady(), nil
+}
+
+// IsServiceNotRouteReady checks the RouteReady status of the service and return false if the service's
+// RouteReady is true.
+func IsServiceNotRouteReady(s *v1alpha1.Service) (bool, error) {
+	result := s.Status.GetCondition(v1alpha1.ServiceConditionRoutesReady)
+	return result.Status == corev1.ConditionFalse, nil
```

Comment:
I have a feeling that `result.Status` is where you are getting your nil dereference. If you look at other code doing this it looks like

```
        rc := service.Status.GetCondition(v1alpha1.ServiceConditionRoutesReady)
        if rc == nil || rc.Status != corev1.ConditionTrue {
                return
        }
```

You should check for a `nil` result before getting status from it. If it is nil you should continue waiting.

Reply:
Ahh that makes sense. Thank you!!